packages/core/etc/simplevm/utils/get_metadata.sh

```diff
@@ -98,32 +100,52 @@ if [ $? -ne 0 ]; then
   exit 1
 fi
 
-  key_not_found_flag=$(echo "$metadata_response" | jq -r '.detail')
 
-  if [ "$key_not_found_flag" == "Key not found" ]; then
-      echo "Key not found in the response."
-      reset_interval
-      exit 1
-  elif [ "$key_not_found_flag" == "Too Many Requests" ]; then
-      echo "Too many requests sent to the server. Extending interval..."
-      extend_interval
-      exit 1
 
+# Check the status code and act accordingly
+if [ "$response_status" -eq 200 ]; then
+  # Successful response; proceed to any further processing if needed
+  # Write the metadata to a temporary file
+  TMP_FILE="${METADATA_OUTPUT_FILE}.tmp"
+  echo "${metadata_response}"| jq > "${TMP_FILE}"
+
+  # Rename the temporary file to the original filename
+  if mv -f "${TMP_FILE}" "${METADATA_OUTPUT_FILE}"; then
+    log_message "Metadata saved to ${METADATA_OUTPUT_FILE}"
+    reset_interval
+    exit 0
+
+  else
+    log_message "Error: Failed to save metadata to ${METADATA_OUTPUT_FILE}"
+    reset_interval
+    exit 1
   fi
 
-# Write the metadata to a temporary file
-TMP_FILE="${METADATA_OUTPUT_FILE}.tmp"
-echo "${metadata_response}"| jq > "${TMP_FILE}"
+elif [ "$response_status" -eq 400 ] || [ "$response_status" -eq 403 ]; then
+    echo "Error: Too many requests sent to the server. Extending interval..."
```

Comment:
403 -> Forbidden so wrong token (or non allowed ip) --> should log other message
400 - > should log other message
429 -> to many requests (is currently missing )